sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3593,10 +3616,69 @@ pi_result piextMemGetNativeHandle(pi_mem Mem, pi_native_handle *NativeHandle) {
 }
 
 pi_result piextMemCreateWithNativeHandle(pi_native_handle NativeHandle,
-                                         pi_mem *Mem) {
-  (void)NativeHandle;
-  (void)Mem;
-  die("piextMemCreateWithNativeHandle: not supported");
+                                         pi_context Context,
+                                         bool ownNativeHandle, pi_mem *Mem) {
+  PI_ASSERT(Mem, PI_INVALID_VALUE);
+  PI_ASSERT(NativeHandle, PI_INVALID_VALUE);
+  PI_ASSERT(Context, PI_INVALID_CONTEXT);
+  PI_ASSERT(Context->Devices.size() > 0, PI_INVALID_CONTEXT);
+
+  try {
+    // Get size of the allocation
+    void *Base;
+    size_t Size;
+    ZE_CALL(zeMemGetAddressRange,
+            (Context->ZeContext, pi_cast<void *>(NativeHandle), &Base, &Size));
```

Comment:
Should we assert that NativeHandle == Base? Or maybe allow it be within [Base; Base+Size)?

Reply:
Added assert.